analytics/property.go

```diff
@@ -0,0 +1,131 @@
+package analytics
+
+// Property ...
+type Property interface {
+	GetKey() string
+	GetValue() interface{}
+}
+
+type stringProperty struct {
+	key   string
+	value string
+}
+
+// StringProperty ...
+func StringProperty(key string, value string) Property {
```

Comment:
So much of this file will be rewritten once Go 1.18 ships with generics 😬

Reply:
Every Go file ever written will be rewritten after generics will be a thing 😛